### DISCUSSION TITLE
Movehub and Cityhub set color and color-distance sensor off and on on long button-press

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#1801] [Q&A] [answered]
For completeness: after firmware flash ran: issue_385.py
issue_385.py:
```python

from pybricks.hubs import ThisHub
from pybricks.tools import wait

hub = ThisHub()

print(hub.system.reset_reason())

wait(100)

hub.system.shutdown()

```
	
Scenario:
- power-on the hub with color and / or color-distance sensor connected
- long-press hub-button and KEEP pressed after power-off
sensors do flip-flop off and on.
- release hub button
sensors go OFF again
movehub goes to **power-ON**
cityhub stays OFF (as it should)

Occurs on:
	Cityhub and movehub at build 3531
	Cityhub at 3.5.0

Does NOT occur on:
	cityhub at build 3527
	technichub at build 3527 and 3531
	primehub at build 3527 and 3531 (with and without USB)

Nothing important I think, what harm does it do to hold a button a bit longer?

note 1: movehub switches **ON** after the long press button is released, on 3.5.0 and build 3531 and 3527

note 2: movehub, cityhub, technichub, primehub: The color / color-distance sensors do NOT re-light on a long-press on build 3527

As stated: nothing important, but I want to document it, so there can be a reference to it.

Bert

**Comment by dlech** [comment]
Build 3527 is attempting to fix these issues, so that is the only build that matters. (or whatever https://github.com/pybricks/pybricks-micropython/pull/265#issuecomment-2322987544 says if we update the pull request - currently build 3527).

> note 1: movehub switches **ON** after the long press button is released, on 3.5.0 and build 3531 and 3527

So this seems like the only thing that is still broken on build 3527.

**Reply by dlech** [reply, marked as answer]
So you can just keep commenting on that pull request until we have all of the kinks worked out.
